### ISSUE TITLE
Cannot post or add category when there are 20+ categories in the site

### ISSUE THREAD

**Issue by pansamolot** [WordPress/gutenberg#20995]
**tl;dr:** *When blog or site has 20+ categories, "Categories" section in Gutenberg is empty and post cannot be saved or published. Happens in Gutenberg only (other Wordpress functionality unaffected) and only on 20+ categories (sites with smaller categories number unaffected).*

Affects: Chrome browser only. Microsoft Edge not affected. Mobile Chrome not affected.

----

**Describe the bug**
Just updated Gutenberg plugin to 7.7.1 and noticed that:

* **post cannot be saved**,
* external preview cannot be generated,
* categories section is empty,
* new category cannot be added.

when particular **blog has more than 20 categories** ("Categories" section spans into 2+ pages).

**20+ categories cause is confirmed**. I have a blog network with 20+ sites, only two of these sites are big enough to have 20+ categories and reported problem appears only in these two sites. All remaining 18 sites, with categories count lesser than twenty works normally.

**Gutenberg source of the cause is confirmed**. No database, WordPress or installation issue. All other functionality (like changing and saving anything in Quick Edit mode) works normally.

Possible causes:

An attempt for Gutenberg to get categories ends in endless 403 responses on JSON request:

![image](https://user-images.githubusercontent.com/60975060/76983669-1508ab80-693e-11ea-80d6-bb194d95e18b.png)

When development version of Gutenberg is disabled and Wordpress reverts to built-in version, the whole "Tags" and "Categories" section is gone and JSON request fails with 403, but when requesting user data instead:

![image](https://user-images.githubusercontent.com/60975060/76983965-8183aa80-693e-11ea-8158-881f9c3fc491.png)

Details and scenario follows:

**A. To reproduce**
1. Go to any blog that has more than 20 categories.
2. Open Gutenberg on any blog post (published, draft, create new).
3. Make any change to post body or post settings to enable 'Save Draft' button.
4. Click 'Save Draft' button.
5. Observe that it turns into 'Saving' message forever and changes to blog post are not saved.

**A. Expected behavior**
Changes to blog post are saved, as usually.

**B. To reproduce**
1. Go to any blog that has more than 20 categories.
2. Open Gutenberg on any blog post (published, draft, create new).
3. Make any change to post body.
4. Click down arrow next to "Desktop" and select "Preview externally".
5. Observe that preview generation hangs indefinitely and preview isn't ever generated.

**B. Expected behavior**
Preview is generated, as usually.

**C. To reproduce**
1. Go to any blog that has more than 20 categories.
2. Open Gutenberg on any blog post (published, draft, create new).
3. Scroll down to 'Categories' section in the right pane.
4. Observe that this section is empty.

**C. Expected behavior**
"Categories" section filled with categories, as usually.

**D. To reproduce**
1. Go to any blog that has more than 20 categories.
2. Open Gutenberg on any blog post (published, draft, create new).
3. Scroll down to 'Categories' section in the right pane.
4. Click on the "Add New Category" link.
5. Provide new category's name into "New Category Name" box.
6. Hit the "Add New Category" button.
7. Observe that new category is never actually created.

**D. Expected behavior**
New category is added, as usually.

**Scenarios C and D can only be reproduced when using development version of Gutenberg**. Only then "Categories" section in Gutenberg is empty, but still visible. When network-disabling development version of Gutenberg and thus reverting blog network back to built-in version of Gutenberg, **the whole "Categories" section vanishes** and these two scenarios cannot be reproduced.

**Screenshots**
Empty "Categories" section (development version of Gutenberg, 20+ categories site):

![image](https://user-images.githubusercontent.com/60975060/76982329-04573600-693c-11ea-9ff2-8685b4e6dc5c.png)

"Categories" section working as expected (less than 20 categories blog):

![image](https://user-images.githubusercontent.com/60975060/76982462-39fc1f00-693c-11ea-80e4-9e2ec736c03e.png)

"Categories" and "Tags" sections gone completely (non-development version of Gutenberg, 20+ categories site):

![image](https://user-images.githubusercontent.com/60975060/76982604-69129080-693c-11ea-80a7-73ecebaf14ca.png)

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise 1903
 - Browser Chrome
 - Version 80.0.3987.132

Other browsers **NOT** affected. Tested in Microsoft Edge.

**Smartphone (please complete the following information):**
**NOT** affected.

**Additional context**
- Version of Gutenberg: 7.7.1 development version + production version on board Wordpress 5.3.2.

**Comment by pansamolot** [comment]
This issue most likely is not a real bug, but rather some issue in my Chrome. I can't replicate this problem in other browsers and in Chrome installed on other PCs.

However, I think it might be interesting to investigate why this particular bug / problem appears in the situation where there are more than 20 categories **only**. There may be some issue related to categories pagination in JSON used for rendering categories box.

I am leaving a decision whether we should or shouldn't close this bug to project admins.

**Comment by noisysocks** [comment]
I wasn't able to reproduce this on a site with 71 categories running 5.5.1 and Gutenberg 9.0.

@pansamolot: Are you using multisite? Do you have any extensions in Google Chrome that might be intercepting requests? Are you able to, in DevTools, see the error message that the REST API is sending back to Gutenberg?

**Comment by talldan** [comment]
Closing due to the lack of reply.
